backstory/src/main/java/com/google/sps/servlets/StoryManager.java

```diff
@@ -0,0 +1,111 @@
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.IdTokenProvider;
+import java.io.IOException;
+import java.util.Scanner;
+import org.json.JSONObject;
+
+public final class StoryManager {
+  /**
+   * Makes a post request with a JSON including GPT2 Parameters
+   */
+  private static HttpResponse makePostRequestGPT2(
+      String serviceUrl, String prompt, int textLength, Double temperature) throws IOException {
+    // Obtain Credentials
+    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+
+    // Validate Credentials
+    if (!(credentials instanceof IdTokenProvider)) {
+      throw new IllegalArgumentException("Credentials are not an instance of IdTokenProvider.");
+    }
+
+    // Generate Authentication Token
+    IdTokenCredentials tokenCredential = IdTokenCredentials.newBuilder()
+                                             .setIdTokenProvider((IdTokenProvider) credentials)
+                                             .setTargetAudience(serviceUrl)
+                                             .build();
+
+    // Configure URL
+    GenericUrl genericUrl = new GenericUrl(serviceUrl);
+
+    // Form Adapter with Authentication token
+    HttpCredentialsAdapter adapter = new HttpCredentialsAdapter(tokenCredential);
+    HttpTransport transport = new NetHttpTransport();
+
+    // Form JSON body using generation parameters
+    String requestBody = "{\"length\": " + textLength
+        + ",\"truncate\": \"<|endoftext|>\", \"prefix\": \"" + prompt
+        + "\", \"temperature\": " + temperature + "}";
+
+    // Build Request with Adapter and JSON Input
+    HttpRequest request = transport.createRequestFactory(adapter).buildPostRequest(
+        genericUrl, ByteArrayContent.fromString("application/json", requestBody));
+    request.getHeaders().setContentType("application/json");
+
+    // Wait until response received
+    request.setConnectTimeout(0);
+    request.setReadTimeout(0);
+    return request.execute();
+  }
+  /**
+   * Returns generated text output for a given prompt, length, and temperature.
+   */
+  public static String generateText(String prompt, int textLength, Double temperature) {
+    // Obtain response from Server POST Request
+    try {
+      HttpResponse outputResponse = makePostRequestGPT2(
+          "https://backstory-text-gen-pdaqhmzgva-uc.a.run.app", prompt, textLength, temperature);
+
+      // Parse response as JSON
+      try {
+        JSONObject jsonObject = new JSONObject(outputResponse.parseAsString());
+        return jsonObject.getString("text");
+      } catch (Exception jsonException) {
+        throw new RuntimeException("Failed to convert repsonse into JSON", jsonException);
+      }
+    } catch (IOException serverException) {
+      throw new RuntimeException("Error with server", serverException);
+    }
+  }
+
+  /**
+   * An executable demonstration of GPT-2 interface.
+   */
+  public static void main(String[] args) {
```

Comment:
Could this "debug" code be separated from other "production" code? Maybe this could be located outside of this package.

Reply:
Done